scalableminds/webknossos
# Avoid sudden dataset reordering in dashboard

This PR fixes #4461 in a rudimentary way. I thought quite a while about this and I think this solution is fine. In the most common use case, the user will open the dashboard and it will simply work (be fast and won't show any sudden changes). If the user wants to open a new dataset (which was just added), they will likely click the "refresh" button, anyway, and then the new dataset will appear. There is the chance that a "soft reload" (instead of having the back-end rescan all directories) would also do the trick, but integrating this into the UI is likely quite confusing. I briefly thought about updating the dataset list with the updated cache content when the mouse is at a good position (e.g., when it hovers over the fresh button), but in the end this is probably too weird.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- it's a bit hard to test which is why I monkey-patched the `getDatasets` api to drop (or not drop) some elements between page refreshs
- during that, I could verify that a change in the dataset list "arrived" on the second page refresh (since a page load will always show the cache contents which get updated when loading the dataset)

### Issues:
- contributes to #4461

------
